src/runtime/sgx/trusted/ecall_registry.cc

```diff
@@ -0,0 +1,82 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file ecall_registry.cc
+ * \brief The global registry of packed functions available via ecall_packed_func.
+ */
+#include <dmlc/logging.h>
+#include <tvm/runtime/registry.h>
+
+namespace tvm {
+namespace runtime {
+namespace sgx {
+
+class ECallRegistry: public Registry {
+ public:
+  explicit ECallRegistry(std::string name) {
+    name_ = name;
+  }
+
+  Registry& set_body(PackedFunc f) {
+     func_ = f;
+     return *this;
+  }
+
+  Registry& set_body(PackedFunc::FType f) {  // NOLINT(*)
+    return set_body(PackedFunc(f));
+  }
+
+  static Registry& Register(const std::string& name, bool override = false) {
+    for (auto& r : exports_) {
+      if (r.name_ == name) {
+        CHECK(override) << "ecall " << name << " is already registered";
+        return r;
+      }
+    }
+    TVM_SGX_CHECKED_CALL(tvm_ocall_register_export(name.c_str()));
```

Comment:
better also pass fid, instead of relying on the order of tvm_ocall_register_export

Reply:
it's single-threaded during initialization so they'll always be the same, but I suppose it helps to be explicit

Reply:
yes, just to be explicit